middleware/kubernetes/handler.go

```diff
@@ -39,37 +41,40 @@ func (k Kubernetes) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.M
 		zone = state.Name()
 	}
 
-	var (
-		records, extra []dns.RR
-		err            error
-	)
-	switch state.Type() {
-	case "A":
-		records, _, err = middleware.A(&k, zone, state, nil, middleware.Options{})
-	case "AAAA":
-		records, _, err = middleware.AAAA(&k, zone, state, nil, middleware.Options{})
-	case "TXT":
-		records, _, err = middleware.TXT(&k, zone, state, middleware.Options{})
-	case "CNAME":
-		records, _, err = middleware.CNAME(&k, zone, state, middleware.Options{})
-	case "PTR":
-		records, _, err = middleware.PTR(&k, zone, state, middleware.Options{})
-	case "MX":
-		records, extra, _, err = middleware.MX(&k, zone, state, middleware.Options{})
-	case "SRV":
-		records, extra, _, err = middleware.SRV(&k, zone, state, middleware.Options{})
-	case "SOA":
-		records, _, err = middleware.SOA(&k, zone, state, middleware.Options{})
-	case "NS":
-		if state.Name() == zone {
-			records, extra, _, err = middleware.NS(&k, zone, state, middleware.Options{})
-			break
+	records, extra, _, err := k.routeRequest(zone, state)
+
+	if k.AnticipatePaths && k.IsNameError(err) {
+		p := k.findPodWithIP(state.IP())
+		if p != nil {
+			name, path, ok := splitSearchPath(zone, state.QName(), p.Namespace)
+			if ok {
```

Comment:
I changed `if p != nil ` to `for p != nil ` which allowed me to reduce indentation in several places with early breaks... return at the end means the loop can only execute once...